extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoHandler.java

```diff
@@ -56,151 +55,140 @@
  * @author Stefan Bußweiler - Added new thing status handling
  */
 
-public class WemoHandler extends BaseThingHandler {
+public class WemoHandler extends BaseThingHandler implements UpnpIOParticipant, DiscoveryListener {
 
     private final Logger logger = LoggerFactory.getLogger(WemoHandler.class);
 
-    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets.newHashSet(WEMO_SOCKET_TYPE_UID,
-            WEMO_INSIGHT_TYPE_UID, WEMO_LIGHTSWITCH_TYPE_UID, WEMO_MOTION_TYPE_UID, WEMO_SENSOR_TYPE_UID);
-    private static String getInsightParamsXML;
-    private static String getRequestXML;
-    private static String setRequestXML;
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets.newHashSet(THING_TYPE_SOCKET, THING_TYPE_INSIGHT,
+            THING_TYPE_LIGHTSWITCH, THING_TYPE_MOTION);
 
-    static {
-        try {
-            getInsightParamsXML = IOUtils.toString(WemoHandler.class
-                    .getResourceAsStream("/org/eclipse/smarthome/binding/wemo/internal/GetInsightParams.xml"));
-            getRequestXML = IOUtils.toString(WemoHandler.class
-                    .getResourceAsStream("/org/eclipse/smarthome/binding/wemo/internal/GetRequest.xml"));
-            setRequestXML = IOUtils.toString(WemoHandler.class
-                    .getResourceAsStream("/org/eclipse/smarthome/binding/wemo/internal/SetRequest.xml"));
-        } catch (Exception e) {
-            LoggerFactory.getLogger(WemoHandler.class).error("Cannot read XML files!", e);
-        }
-    }
+    private Map<String, String> stateMap = Collections.synchronizedMap(new HashMap<String, String>());
+
+    protected final static int SUBSCRIPTION_DURATION = 600;
+
+    private UpnpIOService service;
+    private DiscoveryServiceRegistry discoveryServiceRegistry;
```

Comment:
The DiscoveryServiceRegistry is not meant to be used by bindings.

If I see it right, you only use it to know if your device with a given UDN is available or not - this information should actually be provided by UpnpIOParticipant .onStatusChanged (which you have ignored in the code below).